### ISSUE TITLE
get stucked and crash when view one installed game with too many DLCs over MTP

### ISSUE THREAD

**Issue by lanvent** [rashevskyv/dbi#85]
Hi, I usually use NSCB to check for updates. Lately it's not working. I found the problem was that NSCB failed to read the game list over DBI MTP.  

I opened the game directory one by one over DBI MTP and the MTP would get stuck and crash when opening a game with 323 DLCs. I think that's probably why NSCB failed. I also tested over Tinfoil MTP, the game directory could be opened and dumped well. 

The following is my device info:
Switch revision: Mariko
HOS version: 13.2.0
CFW version: 13.2.0|AMS 1.2.5
DBI version: v350
Tinfoil version: 11.0
System: Windows 10 Home 21H1

And here is the stucked game:
Stucked game: Taiko No Tatsujin Nintendo Switch Version! [B+U2949120+323DLC].nsp
![image](https://user-images.githubusercontent.com/26515464/150650336-885e3478-6be4-4a44-9d0c-6ae76d354d7c.png)

If you need the raw XCI file, I can share it with you through Google Drive or BaiduNetDisk. And If you need more info, please tell me.

Thanks for your attention!

**Comment by lanvent** [comment]
Also ran into another issue over MTP: when deleting a game with many(maybe greater than 10 or more) DLCs, DBI on the Switch finished deleting, but the MTP would get stuck in the progress bar of the delete.

**Comment by duckbill007** [comment]
1. How do you run DBI: applet or title mode?
2. Does dbi also hangs or only windows?
3. Did you tried the same with folder on SD card with many files?

I just tried Taiko game and with Linux on PC side all works as designed.

**Comment by duckbill007** [comment]
Also - can you dump that combined Taiko nsp without entering folder?

**Comment by lanvent** [comment]
Hi, duckbill, I think I've been using the title mode(in album) and I tested the applet mode (press R) , unforunately it's still stuck. At that time DBI looks fine, I can press B to stop MTP responder, only windows hang. 

After hanging for a long time, it looks like the MTP connection is broken because I open another drive, such as drive "External SD card", showing "This folder is empty". And I can't dump Taiko nsp without entering folder.

I installed a virtual machine running Ubuntu 20.04.3 LTS and everything is fine just like you said. I dumped Taiko, copied it(a folder with 326 files) to SDcard over DBI MTP with Linux. Then I read and copied it over DBI MTP with windows, it looks fine.

Finally I tried to reinstall the system Windows 10 Pro 21H2 and everything goes well. I guess the Home Version of Windows 10 has some conservative Settings for MTP. 

Thank you very much for your reply and help. DBI is the best tool I've seen on Switch.
